torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -239,26 +242,31 @@ def _check_order(self, handle: FlatParamHandle, is_training: bool) -> None:
             # Copy entire tensor from D2H once to avoid per element D2H copies
             world_indices = world_indices.cpu()
             # Check that all ranks plan to all-gather the same index parameters
-            for (r1, i1), (r2, i2) in itertools.combinations(
-                (
+            if not torch.distributed._functional_collectives.is_torchdynamo_compiling():
```

Comment:
is this error-checking logic really safe to skip under compile?

in the practical sense, probably.  but technically we're supposed to throw the same set of errors in PT2 as eager.

Reply:
Its not amazing to skip, this also plagued me. One thing we can do is potentially pull this out into an op. It just didn't mean the importance bar for the first MVP.

Reply:
ok.

Reply:
expect_true will work here too